src/middleware/check-authorized.global.ts

```diff
@@ -0,0 +1,15 @@
+import { PageAuthRequirements } from '~/_app/domain/auth'
+import { useAccessToken } from '~/composables/auth/auth-cookies'
+import { useRedirectOnAuth } from '~/_app/use-cases/auth'
+
+export default defineNuxtRouteMiddleware((to, from) => {
+  const redirectOnAuth = useRedirectOnAuth()
+
+  const requirement = to.meta.auth as PageAuthRequirements | undefined
+
+  const accessToken = useAccessToken()
+
+  const isAuthorized = computed(() => !!accessToken.value)
```

Comment:
Create `exists` function with signature like exists<T>(target: T | undefined | null): target is T
`const isAuthorized = computed(() => exists(accessToken.value))`